File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        result=[]
        q=deque()
        q.append(root)



        while q:
            size=len(q)
            level_sum=0
           
            
            for _ in range(size):
                node=q.popleft()
                level_sum += node.val

                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

            result.append(level_sum/size)

        return result
            






            

        bfs(cur,0,count)


        return result

File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var averageOfLevels = function(root) {
    
    if(!root)return 0

    const res=[]
    const queue= [root]

    while (queue.length >0){
        const size =queue.length
        let currentSum=0

        for (let i =0; i<size;i++){
            const node =queue.shift()
            currentSum =currentSum+node.val

            if(node.left) queue.push(node.left)
            if(node.right) queue.push(node.right)

        }

        res.push(currentSum/size)

    }

    return res
};